File: app/src/main/java/com/pluu/webtoon/utils/GlideUtils.java
package com.pluu.webtoon.utils;

import android.graphics.Bitmap;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.TransitionDrawable;
import android.widget.ImageView;

import com.bumptech.glide.load.resource.bitmap.GlideBitmapDrawable;
import com.bumptech.glide.request.target.SquaringDrawable;

/**
 * Glide Image Utils
 * Created by PLUUSYSTEM-NEW on 2016-04-03.
 */
public class GlideUtils {

    /**
     * Get Bitmap
     * @see <a href="http://dalinaum.github.io/android/2015/07/07/retrieving-bitmap-of-glide.html">Base Code Url</a>
     * @param view ImageView
     * @return Bitmap
     */
    public static Bitmap loadGlideBitmap(ImageView view) {
        Bitmap bitmap = null;
        Drawable drawable = view.getDrawable();
        if (drawable instanceof GlideBitmapDrawable) {
            bitmap = ((GlideBitmapDrawable) drawable).getBitmap();
        } else if (drawable instanceof TransitionDrawable) {
            TransitionDrawable transitionDrawable = (TransitionDrawable) drawable;
            int length = transitionDrawable.getNumberOfLayers();
            for (int i = 0; i < length; ++i) {
                Drawable child = transitionDrawable.getDrawable(i);
                if (child instanceof GlideBitmapDrawable) {
                    bitmap = ((GlideBitmapDrawable) child).getBitmap();
                    break;
                } else if (child instanceof SquaringDrawable
                        && child.getCurrent() instanceof GlideBitmapDrawable) {
                    bitmap = ((GlideBitmapDrawable) child.getCurrent()).getBitmap();
                    break;
                }
            }
        } else if (drawable instanceof SquaringDrawable) {
            bitmap = ((GlideBitmapDrawable) drawable.getCurrent()).getBitmap();
        }
        return bitmap;
    }

}

File: app/src/main/java/com/pluu/webtoon/utils/GlideUtils.kt
package com.pluu.webtoon.utils

import android.graphics.Bitmap
import android.graphics.drawable.BitmapDrawable
import android.graphics.drawable.TransitionDrawable
import android.widget.ImageView
import com.bumptech.glide.load.resource.gif.GifDrawable

/**
 * Glide Image Utils
 * Created by pluu on 2017-04-29.
 */
/**
 * Get Bitmap
 * @see [Base Code Url](http://dalinaum.github.io/android/2015/07/07/retrieving-bitmap-of-glide.html)
 * @return Bitmap
 */
fun glideBitmap(imgView: ImageView): Bitmap? {
    val lblDrawable = imgView.drawable
    return when (lblDrawable) {
        is BitmapDrawable -> lblDrawable.bitmap
        is TransitionDrawable -> convertTransitionDrawable(lblDrawable)
        is GifDrawable -> lblDrawable.firstFrame
        else -> null
    }
}

private fun convertTransitionDrawable(drawables: TransitionDrawable): Bitmap? {
    (0 until drawables.numberOfLayers - 1)
            .map { drawables.getDrawable(it) }
            .forEach {
                if (it is BitmapDrawable) {
                    return it.bitmap
                }
            }
    return null
}
